lib/service/services_cli.rb

```diff
@@ -146,14 +146,35 @@ def stop(targets, verbose: false)
         if service.dest.exist?
           ohai "Successfully stopped `#{service.name}` (label: #{service.service_name})"
         elsif service.loaded?
-          kill(service)
+          launchctl_kill(service)
         end
         rm service.dest if service.dest.exist?
       end
     end
 
+    # Stop a service but keep it registered.
+    def kill(targets, verbose: false)
+      targets.each do |service|
+        if !service.pid?
+          opoo "Service `#{service.name}` is not started."
+        elsif service.keep_alive?
+          opoo "Service `#{service.name}` is set to automatically restart and can't be killed."
+        else
+          puts "Killing `#{service.name}`... (might take a while)"
+          if System.systemctl?
+            quiet_system System.systemctl, System.systemctl_scope, "stop", service.service_name
+          elsif System.launchctl?
+            quiet_system System.launchctl, "stop", "#{System.domain_target}/#{service.service_name}"
+          end
+
+          sleep(5) while service.pid?
```

Comment:
Isn't this an infinite loop if the service won't stop?

Reply:
You're absolutely right. I'll just print an error message if the service won't stop.